### PR TITLE
Clean chibios_hwdef.py - flake8 cleanliness

### PR DESCRIPTION
**Pull request by peterbarker**
- make and mark it flake8-clean
 - fix several minor bugs

I've done a configure-all one master and this and verified no changes.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 1]
```diff
@@ -769,47 +773,49 @@ def get_flash_pages_sizes(self):
...
               mcu_series.startswith('STM32G4') or
               mcu_series.startswith('STM32L4')):
-            return [ 2 ] * (self.get_config('FLASH_SIZE_KB', type=int)//2)
+            return [2] * (self.get_config('FLASH_SIZE_KB', type=int)//2)
         else:
             raise Exception("Unsupported flash size MCU %s" % mcu_series)
 
     def get_flash_npages(self):
-        page_sizes = self.get_flash_pages_sizes()
+        pages = self.get_flash_pages_sizes()
```
Comment (magicrub) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:816]: This looks like a bug-fix that changes behavior. What are the impacts of this?

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 2]
```diff
@@ -2036,9 +2023,8 @@ def write_PWM_config(self, f, ordered_timers):
         pwm_out = []
         # start with the ordered list from the dma resolver
         pwm_timers = ordered_timers
-        has_bidir = False
```
Comment (magicrub) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2039]: ```has_bidir = False``` is removed, is that confirmed to not change behavior?

**Review by magicrub — APPROVED** [review]
@peterbarker I'm approving even though I have questions because you addressed them in the main PR description. I just want to re-confirm for sanity's sake.

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2039]: @magicrub - I think its unused - part of the flake8 cleanup which I think should have been in a separate PR

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:816]: Method is never called.

flake8 caught this bug.

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2039]: It's in a separate commit, "minor bug fixes".

No bug fixed materially affects anything except flake8 cleanliness, this rolling these into the one PR.

**Pull request merged**